File: src/main/java/com/leijendary/spring/boot/template/document/SampleTranslationDocument.java
package com.leijendary.spring.boot.template.document;

import static org.springframework.data.elasticsearch.annotations.FieldType.Text;

import org.springframework.data.elasticsearch.annotations.Field;

import lombok.Data;
import lombok.EqualsAndHashCode;

@Data
@EqualsAndHashCode(callSuper = true)
public class SampleTranslationDocument extends LocaleDocument {

    @Field(type = Text, analyzer = "ngram_analyzer", searchAnalyzer = "standard")
    private String name;

    @Field(type = Text, analyzer = "ngram_analyzer", searchAnalyzer = "standard")
    private String description;
}

File: src/main/kotlin/com/leijendary/spring/boot/template/document/SampleTranslationDocument.kt
package com.leijendary.spring.boot.template.document

import com.leijendary.spring.boot.template.core.document.LocaleDocument
import org.springframework.data.elasticsearch.annotations.Field
import org.springframework.data.elasticsearch.annotations.FieldType.Text

class SampleTranslationDocument(
    @Field(type = Text, analyzer = "ngram_analyzer", searchAnalyzer = "standard")
    var name: String,

    @Field(type = Text, analyzer = "ngram_analyzer", searchAnalyzer = "standard")
    var description: String
) : LocaleDocument()